isis/src/base/apps/cam2map/cam2map.cpp

```diff
@@ -20,7 +21,12 @@ namespace Isis {
 
   void cam2map(UserInterface &ui, Pvl *log) {
     // Open the input cube
-    Cube icube(ui.GetFileName("FROM"));
+    Cube icube;
+    CubeAttributeInput inAtt(ui.GetAsString("FROM"));
+    if (inAtt.bands().size() != 0) {
+      icube.setVirtualBands(inAtt.bands());
+    }
```

Comment:
What exactly are virtual bands?

Reply:
virtual bands are how ISIS handles only working with some of the bands in an image. The way that this showed up is running cam2map on a Themis IR image with 3 bands but only wanting to project the first band. So, running cam2map from=themis_image.cub+1 to=just_the_first_band.cub would only map project the first band. This is handled in the API by setting the virtual bands to "1".